lib/sacastats_web/controllers/character_controller.ex

```diff
@@ -0,0 +1,66 @@
+defmodule SacaStatsWeb.CharacterController do
+  use SacaStatsWeb, :controller
+  alias SacaStatsWeb.CharacterView
+
+  def character(conn, %{"character_name" => name, "stat_type" => stat_template_name}) do
+    # Fetch character info from API and DB,
+
+    if String.downcase(stat_template_name) == "lookup" do
+      redirect(conn, to: Routes.character_path(conn, :character, name, "general"))
+    end
+
+    q =
+      PS2.API.Query.new(collection: "character")
+      |> PS2.API.QueryBuilder.term("name.first_lower", String.downcase(name))
+
+    {:ok, %PS2.API.QueryResult{data: body}} = PS2.API.query_one(q)
+    {:ok, %PS2.API.QueryResult{returned: returned}} = PS2.API.query_one(q)
```

Comment:
You can also pattern match on multiple fields in a struct/map at once
```suggestion
    {:ok, %PS2.API.QueryResult{data: body, returned: returned}} = PS2.API.query_one(q)
```

Reply:
The below suggestion changes this too, just wanted to make a note of it :)